calico_cni/calico_cni.py

```diff
@@ -202,7 +202,16 @@ def add(self):
         endpoint = self._provision_veth(endpoint)
         
         # Step 4: Provision / set profile on the created endpoint.
-        self.policy_driver.set_profile(endpoint)
+        try:
+            self.policy_driver.set_profile(endpoint)
+        except policy_drivers.ApplyProfileError, e:
```

Comment:
We need to do more than just release the IP here.  

We've done a few things at this point:
1) Assign IP address
2) Create Endpoint object
3) Provision veth interface

Each of these needs to be un-done (probably in reverse order) if we hit an error.